pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,346 @@
+	}
+
+	secrets, _ := c.getSecretsFromPayload(data.Data)
+	secret := secrets[request.Name]
+
+	if secret == "" {
+		return nil, &APIError{Message: fmt.Sprintf("secret %s for project '%s' and environment '%s' not found", request.Name, request.Project, request.Environment)}
+	}
+
+	return &SecretResponse{Name: request.Name, Value: secrets[request.Name]}, nil
+}
+
+func (c *OnboardbaseClient) GetSecrets(request SecretsRequest) (*SecretsResponse, error) {
+	headers := headers{}
+
+	params := request.buildQueryParams()
+	response, apiErr := c.performRequest("/secrets", "GET", headers, params, httpRequestBody{})
+	if apiErr != nil {
+		return nil, apiErr
+	}
+
+
+	var data secretResponseBody
+	if err := json.Unmarshal(response.Body, &data); err != nil {
+		return nil, &APIError{Err: err, Message: "unable to unmarshal secret payload", Data: string(response.Body)}
+	}
+
+	secrets, _ := c.getSecretsFromPayload(data.Data)
+	return &SecretsResponse{ Secrets: secrets, Body: response.Body}, nil
+}
+
+func (r *SecretsRequest) buildQueryParams() queryParams {
+	params := queryParams{}
+
+	if r.Project != "" {
+		params["project"] = r.Project
+	}
+
+
+	if r.Environment != "" {
+		params["environment"] = r.Environment
+	}
+
+	return params
+}
+
+
+func (r *SecretRequest) buildQueryParams() queryParams {
+	params := queryParams{}
+
+	if r.Project != "" {
+		params["project"] = r.Project
+	}
+
+
+	if r.Environment != "" {
+		params["environment"] = r.Environment
+	}
+
+	return params
+}
+
+func (c *OnboardbaseClient) performRequest(path, method string, headers headers, params queryParams, body httpRequestBody) (*apiResponse, error) {
+	urlStr := c.BaseURL().String() + path
+	reqURL, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, &APIError{Err: err, Message: fmt.Sprintf("invalid API URL: %s", urlStr)}
+	}
+
+	var bodyReader io.Reader
+	if body != nil {
+		bodyReader = bytes.NewReader(body)
+	} else {
+		bodyReader = http.NoBody
+	}
+
+	req, err := http.NewRequest(method, reqURL.String(), bodyReader)
+	if err != nil {
+		return nil, &APIError{Err: err, Message: "unable to form HTTP request"}
+	}
+
+	if method == "POST" && req.Header.Get("content-type") == "" {
+		req.Header.Set("content-type", "application/json")
+	}
+
+	if req.Header.Get("accept") == "" {
+		req.Header.Set("accept", "application/json")
+	}
+	req.Header.Set("user-agent", c.UserAgent)
+	req.Header.Set("api_key", c.OnboardbaseAPIKey)
+
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	query := req.URL.Query()
+	for key, value := range params {
+		query.Add(key, value)
+	}
+	req.URL.RawQuery = query.Encode()
+
+	httpClient := &http.Client{Timeout: 10 * time.Second}
+
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+
+	if !c.VerifyTLS {
+		tlsConfig.InsecureSkipVerify = true
+	}
+
+	httpClient.Transport = &http.Transport{
+		DisableKeepAlives: true,
+		TLSClientConfig:   tlsConfig,
+	}
```

Comment:
this doesn't need to be initialized for each request, instead it can be shared and stored in the `*OnboardbaseClient` struct.